mantidimaging/gui/windows/nexus_load_dialog/presenter.py

```diff
@@ -131,18 +131,20 @@ def _read_rotation_angles(self, image_key: int, before: bool | None = None) -> n
             for sample images are being read.
         :return: A numpy array of the rotation angles or None if all angles provided are 0.
         """
-        assert self.image_key_dataset is not None
+        assert self.image_key_dataset is not None and self.rotation_angles is not None
         if before is None:
             rotation_angles = self.rotation_angles[np.where(self.image_key_dataset[...] == image_key)]
         else:
             first_sample_image_index = np.where(self.image_key_dataset == 0)[0][0]
             if before:
-                rotation_angles = self.rotation_angles[:first_sample_image_index][np.where(
-                    self.image_key_dataset[:first_sample_image_index] == image_key)]
+                image_key_array = np.array(self.image_key_dataset)[:first_sample_image_index]
+                rotation_angles = np.array(
+                    self.rotation_angles)[:first_sample_image_index][np.where(image_key_array == image_key)]
             else:
-                rotation_angles = self.rotation_angles[first_sample_image_index:][np.where(
-                    self.image_key_dataset[first_sample_image_index:] == image_key)]
+                rotation_angles = np.array(self.rotation_angles)[first_sample_image_index:][np.where(
+                    np.array(self.image_key_dataset)[first_sample_image_index:] == image_key)]
 
+        rotation_angles = np.array(rotation_angles)
```

Comment:
Are these `np.array()`s still needed? between the type annotation and the assert I think that should be safe.

Reply:
These can now be removed 👍